samples/drivers/adc/adc_sequence/src/main.c

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) 2024 Centro de Inovacao EDGE
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/drivers/adc.h>
+#include <zephyr/kernel.h>
+
+/* Macro for building an array of configurations. */
+#define ADC_CHANNEL_CONFIG_DT_AND_COMMA(node_id) ADC_CHANNEL_CFG_DT(node_id),
+
+/* ADC node from the devicetree. */
+#define ADC_NODE DT_ALIAS(adc)
+
+/* Get the ADC resolution from one of its children. */
+#define ADC_RESOLUTION DT_PROP(DT_CHILD(ADC_NODE, channel_0), zephyr_resolution)
+
+/* Number of samples to be made on the sequence. */
+#define SEQUENCE_SAMPLES 5
+
+/* Enumerate channels for convenience. */
+enum sequence_channels {
+	CHANNEL_A = 0,
+	CHANNEL_B,
+
+	CHANNEL_COUNT,
```

Comment:
why hardcode the number of channels?

Reply:
what I mean is, can't you count the number of channels from DT, and set the sequence.channels using a for loop

Reply:
Gotcha, I was thinking on a sample with fixed number of channels, throughout the boards, but this seems really better. Updated the initialization to account for that.